ormar/models/metaclass.py

```diff
@@ -195,6 +196,29 @@ def verify_constraint_names(
             )
 
 
+def get_constraint_copy(
+    value: ColumnCollectionConstraint,
+) -> Union[UniqueColumns, IndexColumns, CheckColumns]:
+    """
+    Copy the constraint and unpacking it's values
+
+    :raises ValueError: if non subclass of ColumnCollectionConstraint
+    :param value: an instance of the ColumnCollectionConstraint class
+    :type value: Instance of ColumnCollectionConstraint child
+    :return: copy ColumnCollectionConstraint ormar constraints
+    :rtype: Union[UniqueColumns, IndexColumns, CheckColumns]
+    """
+
+    if isinstance(value, sqlalchemy.UniqueConstraint):
+        return UniqueColumns(*value._pending_colargs)  # pragma: no cover
+    elif isinstance(value, sqlalchemy.Index):
+        return IndexColumns(*value._pending_colargs)  # pragma: no cover
+    elif isinstance(value, sqlalchemy.CheckConstraint):
+        return CheckColumns(value.sqltext)  # pragma: no cover
+
+    raise ValueError(f"{value} Must be a ColumnCollectionConstraint.")
```

Comment:
This can be re-written as:

```python
def get_constraint_copy(
    constraint: ColumnCollectionMixin # I think index is not a constraint in sqlalchemy and Mixin is the least common part?
) -> Union[UniqueColumns, IndexColumns, CheckColumns]::
    constraints = {
        sqlalchemy.UniqueConstraint: lambda x: UniqueColumns(*x._pending_colargs),
        sqlalchemy.Index: lambda x: IndexColumns(*x._pending_colargs),
        sqlalchemy.CheckConstraint: lambda x: CheckColumns(x.sqltext),
    }
    checks = (key if isinstance(constraint, key) else None for key in constraints)
    target_class = next((target for target in checks if target is not None), None)
    constructor = constraints.get(target_class)
    if not constructor:
        raise ValueError(f"{constraint} must be a ColumnCollectionMixin!")
    return constructor(constraint)
```

It seems more complicated but it has number of advantages.
1. You can see straight away what are all your options in constraints
2. You can easily expand by adding new entries to constraints dictionaries without changing anything else
3. You avoid complicated if/else statements
4. You avoid pragma no cover comments

Reply:
Yes, that's right, it was perfect
Thank you for your guidance